Plugin/Quote/UpdateQuote.php

```diff
@@ -0,0 +1,57 @@
+<?php
+declare(strict_types=1);
+
+namespace Amwal\Payments\Plugin\Quote;
+
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Model\QuoteManagement;
+use Amwal\Payments\Model\Checkout\AmwalCheckoutAction;
+use Amwal\Payments\Model\Config\Checkout\ConfigProvider;
+use Magento\Store\Model\StoreManagerInterface;
+use Amwal\Payments\Model\Config;
+
+class UpdateQuote
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * Constructor
+     *
+     * @param StoreManagerInterface $storeManager
+     * @param Config $config
+     */
+    public function __construct(StoreManagerInterface $storeManager, Config $config)
+    {
+        $this->storeManager = $storeManager;
+        $this->config = $config;
+    }
+
+    /**
+     * Before submit quote plugin
+     * @param QuoteManagement $subject
+     * @param Quote $quote
+     */
+    public function beforeSubmit(QuoteManagement $subject, Quote $quote)
+    {
+        $store = $this->storeManager->getStore();
```

Comment:
I believe this logic is applied to all quotes now. We should probably add some logic so it's limited to quotes for Amwal only